### ISSUE TITLE
Using Textmate rules for highlighting Rascal code in Asciiidoctor-based documentation?

### ISSUE THREAD

**Issue by PaulKlint** [usethesource/rascal-syntax-highlighting#9]
This is a question rather than an issue.

At the moment we are working on new infrastructure for our Rascal documentation that is based on Asciidoctor (see http://asciidoctor.org). Asciidoctor supports highlighting in various ways (see http://asciidoctor.org/docs/user-manual/#source-code-blocks). Now the question is: can we convert the nice Textmate rules under development by @ahmadsalim to input for any of the supported systems by Asciidoctor:
* http://coderay.rubychan.de
* https://highlightjs.org
* https://github.com/google/code-prettify
* http://pygments.org

I would prefer to have less Ruby/Python and more Javascript in the solution we will adopt.

Any thoughts?

**Comment by ahmadsalim** [comment]
Maybe https://github.com/gjtorikian/pointillist could be useful?
I have not heavily investigated it yet, but it may do what is needed.

**Comment by ahmadsalim** [comment]
I have tried searching around for more, and unfortunately I could not find anything more directly useful.

Interestingly, it seems that AsciiDoctor itself is written in Ruby and translated to JavaScript via Opal (http://opalrb.org/), so maybe that is an alternate solution, but I am unsure how well it would work.

**Comment by PaulKlint** [comment]
You are completely right about the implementation of AsciiDoctor. At the moment we are using (only at build time to avoid shipping JRuby in the Rascal jar) AsciiDoctorJ which is a Java wrapper that uses Ruby inside (via Opal as you mentioned), see https://github.com/asciidoctor/asciidoctorj.

In the future we may also want to provide documentation generation facilities at runtime and that would change the above situation.

I am currently focussing on the compiled Rascal REPL, but will come back to this issue asap.

**Comment by ahmadsalim** [comment]
I might have found another more palatable alternative.
One could try [translating](http://flight-manual.atom.io/hacking-atom/sections/converting-from-textmate/) the TextMate grammar to an Atom one, and then use the Atom javascript based [Highlights](https://github.com/atom/highlights) solution.
However, the latter seems to rely on NodeJS, so I am unsure how one could incorporate it inside the browser.

**Comment by ahmadsalim** [comment]
So I have made a new PR #13 which converts the grammar to an Atom compatible one using the proposed link, so that would at least solve one part of the issue I hope.
